src/webgpu/api/validation/encoding/encoder_open_state.spec.ts

```diff
@@ -0,0 +1,383 @@
+          }
+          break;
+        case 'insertDebugMarker':
+          {
+            encoder.insertDebugMarker('marker');
+          }
+          break;
+        case 'pushDebugGroup':
+        case 'popDebugGroup':
+          {
+            encoder.pushDebugGroup('initializeWithStoreOp');
+            encoder.popDebugGroup();
+          }
+          break;
+        case 'writeTimestamp':
+          {
+            const querySet = t.device.createQuerySet({ type: 'timestamp', count: 1 });
+            encoder.writeTimestamp(querySet, 0);
+          }
+          break;
+        case 'resolveQuerySet':
+          {
+            const querySet = t.device.createQuerySet({ type: 'occlusion', count: 1 });
+            encoder.resolveQuerySet(querySet, 0, 1, dstBuffer, 0);
+          }
+          break;
+        default:
+          unreachable();
+      }
+    }, finishBeforeCommand);
+
+    if (!finishBeforeCommand) encoder.finish();
+  });
+
+g.test('render_pass_commands')
+  .params(u =>
+    u
+      .combine('command', kRenderPassEncoderCommands)
+      .beginSubcases()
+      .combine('finishBeforeCommand', [false, true])
+  )
+  .fn(t => {
+    const { command, finishBeforeCommand } = t.params;
+
+    const querySet = t.device.createQuerySet({ type: 'occlusion', count: 1 });
+    const encoder = t.device.createCommandEncoder();
+    const renderPass = beginRenderPassWithQuerySet(t, encoder, querySet);
+
+    if (finishBeforeCommand) {
+      renderPass.end();
+      encoder.finish();
+    }
+
+    t.expectValidationError(() => {
+      switch (command) {
+        case 'setViewport':
+          {
+            const kNumTestPoints = 8;
+            const kViewportMinDepth = 0;
+            const kViewportMaxDepth = 1;
+            renderPass.setViewport(0, 0, kNumTestPoints, 0, kViewportMinDepth, kViewportMaxDepth);
+          }
+          break;
+        case 'setScissorRect':
+          {
+            renderPass.setScissorRect(0, 0, 0, 0);
+          }
+          break;
+        case 'setBlendConstant':
+          {
+            renderPass.setBlendConstant({ r: 1.0, g: 1.0, b: 1.0, a: 1.0 });
+          }
+          break;
+        case 'setStencilReference':
+          {
+            renderPass.setStencilReference(0);
+          }
+          break;
+        case 'beginOcclusionQuery':
+        case 'endOcclusionQuery':
+          {
+            renderPass.beginOcclusionQuery(0);
+            renderPass.endOcclusionQuery();
+          }
+          break;
+        case 'executeBundles':
+          {
+            const bundleEncoder = t.device.createRenderBundleEncoder({
+              colorFormats: ['rgba8unorm'],
+            });
+            const bundle = bundleEncoder.finish();
+            renderPass.executeBundles([bundle]);
+          }
+          break;
+        default:
+          unreachable();
+      }
+    }, finishBeforeCommand);
+
+    if (!finishBeforeCommand) {
+      renderPass.end();
+      encoder.finish();
+    }
```

Comment:
Since we're just trying to test for validation errors coming directly from the encoding commands, I don't think we really need to end/finish at all here. See also comment on line 370

Reply:
Ah, I see. I just thought that it would be good to end/finish the pass and encoder explicitly. Removed.